tests/ad_common_tests/step-44-helper_scalar_01.h

```diff
@@ -2107,8 +2107,9 @@ namespace Step44
           std::cout << " SLV " << std::flush;
           if (parameters.type_lin == "CG")
             {
-              const int solver_its = tangent_matrix.block(u_dof, u_dof).m() *
-                                     parameters.max_iterations_lin;
+              const int solver_its = static_cast<int>(
+                static_cast<double>(tangent_matrix.block(u_dof, u_dof).m()) *
+                parameters.max_iterations_lin);
```

Comment:
Why does this make sense? Isn't the original product between two integers? Or do you get overflow?

Reply:
`parameters.max_iterations_lin` is a double

To be honest, I should have simplified this entire test before writing it, but I was super lazy.

Reply:
I tend to think that that's the bug: `max_iterations_lin` ought to be an integer of some kind.

Reply:
I think that `max_iterations_lin` was a mistake in the first place: I believe that systems preconditioned using Gauss-Seidel methods should converge in `n_rows` iterations or less. I don't know  why I added this parameter. I'm going to remove it now (in the tests, not step-44 itself).